lib/spaceship/portal/device.rb

```diff
@@ -121,7 +121,7 @@ def find(device_id, mac: false)
         # @return (Device) Find a device based on the UDID of the device. nil if no device was found.
         def find_by_udid(device_udid, mac: false)
           all(mac: mac).find do |device|
-            device.udid == device_udid
+            device.udid.downcase == device_udid.downcase
```

Comment:
This is fine by me as-is - one thing to note is that this creates 2 throw-away strings to do the comparison